### PR TITLE
tiltfile: parallelize docker-compose config calls for faster manifest load [ch1201]

### PR DESCRIPTION
**Pull request by maiamcc**
this will save us .4s per dc.yml when loading manifests.

in particular, a big gain for tests:
```
$ # on master
$ go test github.com/windmilleng/tilt/internal/engine -count=1 && go test github.com/windmilleng/tilt/internal/tiltfile -count=1
ok  	github.com/windmilleng/tilt/internal/engine	15.682s
ok  	github.com/windmilleng/tilt/internal/tiltfile	14.437s

$ # on this branch
$ go test github.com/windmilleng/tilt/internal/engine -count=1 && go test github.com/windmilleng/tilt/internal/tiltfile -count=1
ok  	github.com/windmilleng/tilt/internal/engine	10.958s
ok  	github.com/windmilleng/tilt/internal/tiltfile	8.815s
```
So that's a savings of c. 10s per test suite run.

### CODE REVIEW

File: internal/tiltfile/docker_compose.go [hunk 1]
```diff
@@ -293,6 +282,56 @@ func parseDCConfig(ctx context.Context, dcc dockercompose.DockerComposeClient, c
 	return services, nil
 }
 
+func getConfigAndServiceNames(ctx context.Context, dcc dockercompose.DockerComposeClient,
+	configPath string) (conf dcConfig, svcNames []string, err error) {
+	// calls to `docker-compose config` take a bit, and we need two,
+	// so do them in parallel to make things faster
+	configGot := make(chan error, 1)
```
Comment (jazzdan) [inline internal/tiltfile/docker_compose.go]: Could we make these unbuffered channels? I know it's guaranteed that each channel will only receive one thing but I find the behavior of unbuffered channels easier to reason about.

File: internal/tiltfile/docker_compose.go [hunk 2]
```diff
@@ -293,6 +282,56 @@ func parseDCConfig(ctx context.Context, dcc dockercompose.DockerComposeClient, c
...
+	servicesGot := make(chan error, 1)
+
+	go func() {
```
Comment (nicks) [inline internal/tiltfile/docker_compose.go]: the idiomatic way to do this is with an errgroup
https://godoc.org/golang.org/x/sync/errgroup#ex-Group--Parallel

File: internal/tiltfile/docker_compose.go [hunk 3]
```diff
@@ -293,6 +282,56 @@ func parseDCConfig(ctx context.Context, dcc dockercompose.DockerComposeClient, c
...
+		}
+		close(servicesGot)
+	}()
+
+	select {
+	case err := <-configGot:
+		if err != nil {
+			return conf, svcNames, err
+		}
+	case <-time.After(time.Second * 2):
```
Comment (nicks) [inline internal/tiltfile/docker_compose.go]: it seems weird to have a timeout here when there wasn't a timeout before? also 2s is shorter than some "normal" times i've seen

File: internal/tiltfile/docker_compose.go (hunk 2 above)
Comment (maiamcc) [reply internal/tiltfile/docker_compose.go]: oh yeah, right! will do

File: internal/tiltfile/docker_compose.go (hunk 3 above)
Comment (maiamcc) [reply internal/tiltfile/docker_compose.go]: yeahhh i just got antsy about leaving a `<- chan` in the middle of the code? but you're right that the previous behavior wasn't any safer in that regard.

**Pull request merged**